packages/iobridge-commons/src/message/cross-chain-message.ts

```diff
@@ -0,0 +1,66 @@
+import { recoverPersonalSignature } from '@metamask/eth-sig-util';
+import { Script } from '@ckb-lumos/lumos';
+
+interface EthereumRpc {
+  (payload: { method: 'personal_sign'; params: [string /* from */, string /* message */] }): Promise<string>;
+}
+
+export interface EthereumProvider {
+  selectedAddress: string;
+  isMetaMask?: boolean;
+  enable: () => Promise<string[]>;
+  addListener: (event: 'accountsChanged', listener: (addresses: string[]) => void) => void;
+  removeEventListener: (event: 'accountsChanged', listener: (addresses: string[]) => void) => void;
+  request: EthereumRpc;
+}
+
+export class ValidateError extends Error {}
+
+export type CrossChainMessage = { txHash: string; toAddress: string };
+
+export function validateCrossChainMessage(message: string): void {
+  try {
+    const messageObj = JSON.parse(message);
+
+    if (typeof messageObj.txHash !== 'string') {
+      throw new Error();
+    }
+
+    if (typeof messageObj.toAddress !== 'string') {
+      throw new Error();
+    }
+  } catch (_) {
+    throw new ValidateError('illegal message, please check message');
+  }
+}
+
+export function verifyCrossChainMessage(payload: {
+  depositedFromPWLock: Script;
+  signature: string;
+  message: string;
+}): boolean {
```

Comment:
we only need to know if the message is signed from an account, as for how the CKB lock is resolved to an Eth address we can implement it in another function, trying to ensure the [principle of least knowledge](https://en.wikipedia.org/wiki/Law_of_Demeter)

```ts
function verifyCrossChainMessage(payload: {
  expectedAddress: string, 
  signature: string, 
  message: CrossChainMessage 
}): boolean 
```